File: sample/src/main/java/pl/droidsonroids/gif/sample/GifSelectorDrawable.java
package pl.droidsonroids.gif.sample;

import android.content.res.Resources;
import android.content.res.XmlResourceParser;
import android.graphics.drawable.StateListDrawable;
import android.support.annotation.DrawableRes;
import android.util.AttributeSet;

import org.xmlpull.v1.XmlPullParser;
import org.xmlpull.v1.XmlPullParserException;

import java.io.IOException;

import pl.droidsonroids.gif.GifDrawable;

public class GifSelectorDrawable extends StateListDrawable {

	private static final String NAMESPACE = "http://schemas.android.com/apk/res/android";

	@Override
	public void inflate(Resources r, XmlPullParser parser, AttributeSet attrs, Resources.Theme theme) throws XmlPullParserException, IOException {
		final XmlResourceParser resourceParser = (XmlResourceParser) parser;
		resourceParser.setFeature(XmlPullParser.FEATURE_PROCESS_NAMESPACES, true);

		int eventType = resourceParser.getEventType();
		do {
			if (eventType == XmlPullParser.START_TAG && "item".equals(resourceParser.getName())) {
				@DrawableRes
				final int resourceId = resourceParser.getAttributeResourceValue(NAMESPACE, "drawable", 0);
				final boolean state_pressed = resourceParser.getAttributeBooleanValue(NAMESPACE, "state_pressed", false);
				final int[] stateSet = state_pressed ? new int[]{android.R.attr.state_pressed} : new int[0];
				addState(stateSet, GifDrawable.createFromResource(r, resourceId));
			}
			eventType = resourceParser.next();
		} while (eventType != XmlPullParser.END_DOCUMENT);
	}
}

File: sample/src/main/java/pl/droidsonroids/gif/sample/GifSelectorDrawable.kt
package pl.droidsonroids.gif.sample

import android.content.res.Resources
import android.content.res.XmlResourceParser
import android.graphics.drawable.StateListDrawable
import androidx.annotation.DrawableRes
import android.util.AttributeSet
import org.xmlpull.v1.XmlPullParser
import org.xmlpull.v1.XmlPullParserException
import pl.droidsonroids.gif.GifDrawable
import java.io.IOException

private const val NAMESPACE = "http://schemas.android.com/apk/res/android"

class GifSelectorDrawable : StateListDrawable() {

	@Throws(XmlPullParserException::class, IOException::class)
	override fun inflate(r: Resources, parser: XmlPullParser, attrs: AttributeSet, theme: Resources.Theme?) {
		val resourceParser = parser as XmlResourceParser
		resourceParser.setFeature(XmlPullParser.FEATURE_PROCESS_NAMESPACES, true)

		var eventType = resourceParser.eventType
		do {
			if (eventType == XmlPullParser.START_TAG && "item" == resourceParser.name) {
				@DrawableRes
				val resourceId = resourceParser.getAttributeResourceValue(NAMESPACE, "drawable", 0)
				val statePressed = resourceParser.getAttributeBooleanValue(NAMESPACE, "state_pressed", false)
				val stateSet = if (statePressed) intArrayOf(android.R.attr.state_pressed) else IntArray(0)
				addState(stateSet, GifDrawable.createFromResource(r, resourceId))
			}
			eventType = resourceParser.next()
		} while (eventType != XmlPullParser.END_DOCUMENT)
	}
}
